dot/core/helpers_test.go

```diff
@@ -148,27 +148,11 @@ func createTestService(t *testing.T, genesisFilePath string,
 	return service, encodedExtrinsic
 }
 
-// NewTestService creates a new test core service
-func NewTestService(t *testing.T, cfg *Config) *Service {
+// newTestService creates a new test core service
+func newTestService(t *testing.T, cfg Config) *Service {
```

Comment:
There are arguments for and against using pointers.
- using a reference is obviously cheaper resource-wise, compared to passing a value, while
- values are safer. no nil pointer error, saves you from race conditions, etc.
- 
But, more often than not, people use pointers or values based on what seems more convenient at the time and based on their personal preferences. 

Unless there is a strong reason for changing such code, I would prefer to have it stay as it is.

Reply:
> using a reference is obviously cheaper resource-wise, compared to passing a value

Not at all. Passing by value is most of the time faster and puts less pressure on the GC. A good rule of thumb is to pass by pointers >= 80 bytes structs. Now in this particular case, the config struct has 8 pointers (8x8 bytes) + log level (1 byte) + map (I think it's ~24B) = 89 bytes, so technically yes passing by pointer would yield a small performance advantage. BUT practically speaking, passing a config (even in production code) to a service is not something executed often so the optimisation is very useless. Worth dumbing down the code to use values and avoid nil pointer checks/panics.

> saves you from race conditions

I think you meant accidental mutation, which is also debatable, since often there are pointers in struct such a maps or slices, so it doesn't shield it completely either.

Sent a message on Slack about this to get alignment across the team :wink: